pkg/status/writer_wrapper.go

```diff
@@ -0,0 +1,20 @@
+package status
+
+import (
+	"context"
+
+	"bigtable-autoscaler.com/m/v2/pkg/interfaces"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type writerWrapper struct {
+	statusWriter ctrlclient.StatusWriter
+}
+
+func (s *writerWrapper) Update(ctx context.Context, obj runtime.Object, opts ...ctrlclient.UpdateOption) error {
+	return s.statusWriter.Update(ctx, obj, opts...)
```

Comment:
If the interface you're wrapping is identical, is a wrapper really necessary? Seems that the wrapped `Update` already implements our `WriterWrapper` interface.

Reply:
In golang, if you implement all methods from an interface, you implement the interface itself:

https://play.golang.org/p/aqJUnXjcTyb